### ISSUE TITLE
[75] Vision-Language Pre-Training with Triple Contrastive Learning

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#104]
[rosinality](https://github.com/rosinality)'s comment
```
vision-language 모델들이 서로 다른 modality의 alignment에 대해서는 많이 탐색했는데
개별 modal에 대한 representation 학습은 없지 않았나 하는 아이디어. 
그래서 cross modal align + intra modal align을 기본으로 하고 
global-local mi, image-text matching, mlm을 끼얹은 vision-language model을 만들었음
```

[paper](https://arxiv.org/pdf/2202.10401.pdf)
[code](https://github.com/uta-smile/TCL)

# TCL (Triple Contrastive Loss)
![image](https://user-images.githubusercontent.com/16400591/155425118-9830e3d4-8ce9-45c9-957d-fd48a6c6fcec.png)

[Align before Fuse](https://arxiv.org/pdf/2107.07651.pdf) 주장에 따르면, Text Encoder, Image Encoder 다음에 Fusion Encoder 를 붙이면 더 좋다.
그 구조는 그대로 냅두고, 좀 더 발전된 형태의 alignment 를 제시한다.

Image 는 I1, I2 2개로 나뉘어서 각각 img encoder, momentum img encoder 에 들어감
Text 는 별다른 변형 없이 text encoder, momentum text encoder 에 들어감

Negative sample 은 배치 내에서 구성

## Cross-Modal Alignment (CMA)
CMA 의 목적은 image 와 text embedding 이 가까워지도록 하는 것.
tilde 가 붙어 있는 애들은 negative sample 을 의미한다.

loss 는 InfoNCE
![image](https://user-images.githubusercontent.com/16400591/155429261-bd2c2f1a-c8dd-40b1-9458-cd644c250edf.png)
![image](https://user-images.githubusercontent.com/16400591/155429477-19aee78c-7615-4b21-b4eb-d023e80947f9.png)
![image](https://user-images.githubusercontent.com/16400591/155429496-717cda58-21d4-4585-bd63-1bbe178d0ac0.png)

## Intra-Modal Contrastive (IMC)
요 녀석은, augmented view 더라도 같은 embedding 으로 모아주는 역할을 함
![image](https://user-images.githubusercontent.com/16400591/155431789-4794cda3-16f1-498b-9244-ff2f124df5f4.png)

## Local MI Maximization (LMI)
global representation과 모든 local region 사이의 MI 를 늘리기 위하는 것.
(예를 들어, image patch CLS와 text token 들. 그림 참고.)

momentum 의 local feature 들이 online net 의 CLS 와 가까워짐.
![image](https://user-images.githubusercontent.com/16400591/155431728-0a8ed62c-4c99-4791-81d9-d716632df665.png)

## Image-Text Matching (ITM)
(img, text) 가 positive (1) 인지 negative (1) 인지 binary classification
![image](https://user-images.githubusercontent.com/16400591/155444172-aff00fe5-4ad9-4bae-aec1-3a065a90a2e5.png)

## Masked Language Modeling (MLM)
BERT loss 와 동일
![image](https://user-images.githubusercontent.com/16400591/155443956-d854eb13-a32c-4eae-88ff-314c33006011.png)

## 최종 Loss
![image](https://user-images.githubusercontent.com/16400591/155443884-b42e1cc0-f013-4c4d-8f31-2961e2fe857c.png)

# Result
## zero-shot retrieval
![image](https://user-images.githubusercontent.com/16400591/155432011-edc79131-d7ea-472b-9f66-e52fbdda9ae3.png)

## finetuned retrieval
![image](https://user-images.githubusercontent.com/16400591/155432082-45acaa92-46a7-4782-b46c-6348b1888be2.png)

## VL Tasks
![image](https://user-images.githubusercontent.com/16400591/155443231-ab14f7e7-aefa-4131-92f7-a76862e0c8f5.png)

## Ablations
![image](https://user-images.githubusercontent.com/16400591/155443340-9af32d51-cf8f-489f-964d-b51490907feb.png)
